open_gpt/serve/gateway.py

```diff
@@ -102,6 +102,76 @@ async def generate(payload: GenerateRequest = Body(...)):
                             },
                         )
 
+            @app.api_route(path='/generate_stream', methods=['POST'])
+            async def generate_stream(payload: GenerateRequest = Body(...)):
+                """Generate text from a prompt in streaming."""
+
+                import asyncio
+
+                from sse_starlette.sse import EventSourceResponse
+
+                STREAM_DELAY = 0.1  # second
+                RETRY_TIMEOUT = 150  # milisecond
+
+                parameters = payload.dict(
+                    exclude_unset=True,
+                    exclude_none=True,
+                    exclude_defaults=True,
+                    exclude={'prompt'},
+                )
+
+                async def event_generator():
+                    print(f"===> in event_generator")
+                    while True:
+                        # Checks for new generated token and return them to client if any
+                        async for docs, error in self.streamer.stream(
+                            docs=input_docs,
+                            exec_endpoint='/generate_stream',
+                            parameters=parameters,
+                        ):
+                            # if error:
+                            #     return JSONResponse(
+                            #         status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
+                            #         content={'message': error.name},
+                            #     )
+                            # else:
+                            print(f"===> in streamer")
+                            input_docs[0].tags['past_key_values'] = docs[0].tags.get(
+                                'past_key_values'
+                            )
+
+                            print(
+                                f"===> output docs tags contains: {list(docs[0].tags.keys())}"
+                            )
+                            print(
+                                f"===> output input_docs tags contains: {list(input_docs[0].tags.keys())}"
+                            )
+                            yield {
+                                "event": "new_message",
+                                "id": "message_id",
+                                "retry": RETRY_TIMEOUT,
+                                "data": str(
+                                    {
+                                        "generated_text": docs[0].tags.get(
```

Comment:
Issue here: docs only contains `past_key_values` and `prompt` in tags field

Reply:
however, the executor returns `generated_text` , `usage`, `finish_reason`

Reply:
"data": json.dumps(docs[0].tags)